src/category_theory/arrow.lean

```diff
@@ -51,6 +51,9 @@ def mk {X Y : T} (f : X ⟶ Y) : arrow T :=
   right := Y,
   hom := f }
 
+theorem mk_inj {T} [category T] (A B : T) (f g : A ⟶ B) : arrow.mk f = arrow.mk g → f = g :=
+by rintro ⟨⟩; refl
```

Comment:
This goes against the mathlib naming guide, that says `_inj` lemmas should be `iff`s. Untested, but:
```suggestion
theorem mk_injective {T} [category T] (A B : T) : function.injective (arrow.mk : (A ⟶ B) → arrow T) :=
by rintro f g ⟨⟩; refl

theorem mk_inj {T} [category T] (A B : T) {f g : A ⟶ B} : arrow.mk f = arrow.mk g ↔ f = g :=
mk_injective.eq_iff A B
```